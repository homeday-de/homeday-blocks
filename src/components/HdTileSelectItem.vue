<template>
  <button
    class="tile-select-item"
    type="button"
    :class="{
      'tile-select-item--selected': isSelected,
    }"
    @click="emitInput"
    @keydown.enter.space.prevent="emitInput"
  >
    {{ text || value }}
  </button>
</template>

<script>
export default {
  name: 'HdTileSelectItem',
  model: {
    prop: 'model',
  },
  props: {
    model: {
      type: [String, Number, Boolean],
      default: '',
    },
    value: {
      type: [String, Number, Boolean],
      required: true,
    },
    text: {
      type: String,
      default: '',
    },
  },
  computed: {
    isSelected() {
      return this.model === this.value;
    },
  },
  methods: {
    emitInput() {
      this.$emit('input', this.value);
    },
  },
};
</script>

<style lang="scss" scoped>
@import "homeday-blocks/src/styles/mixins.scss";

.tile-select-item {
  cursor: pointer;
  width: 100%;
  background-color: white;
  border: 0;
  padding: $inset-m;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  color: inherit;

  &:hover {
    color: $secondary-color;
    box-shadow: 0 0 5px 1px rgba(0, 0, 0, .4);
    z-index: 2;
  }

  &--selected {
    background-color: getShade($secondary-color, 110);
    color: white;
    box-shadow: 0 0 5px 1px rgba(0, 0, 0, .4);
    z-index: 1;

    &:hover {
      color: white;
    }
  }
}
</style>
