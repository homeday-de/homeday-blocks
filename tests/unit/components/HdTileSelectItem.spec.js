import { wrapperFactoryBuilder } from 'tests/unit/helpers';
import HdTileSelectItem from '@/components/HdTileSelectItem.vue';

const model = 1;
const value = 2;
const wrapperFactory = wrapperFactoryBuilder(HdTileSelectItem, {
  props: {
    model,
    value,
  },
});

describe('HdTileSelectItem', () => {
  it('renders component not selected', () => {
    const wrapper = wrapperFactory();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders component selected', () => {
    const wrapper = wrapperFactory({
      props: {
        value: model,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders component with text as display', () => {
    const text = 'Two';
    const wrapper = wrapperFactory({
      props: { text },
    });

    expect(wrapper.text()).toBe(text);
  });

  it('emits value on click', () => {
    const wrapper = wrapperFactory();

    wrapper.trigger('click');

    expect(wrapper.emitted().input[0][0]).toBe(value);
  });

  it('emits value on keydown space', () => {
    const wrapper = wrapperFactory();

    wrapper.trigger('keydown.space');

    expect(wrapper.emitted().input[0][0]).toBe(value);
  });

  it('emits value on keydown enter', () => {
    const wrapper = wrapperFactory();

    wrapper.trigger('keydown.enter');

    expect(wrapper.emitted().input[0][0]).toBe(value);
  });
});
