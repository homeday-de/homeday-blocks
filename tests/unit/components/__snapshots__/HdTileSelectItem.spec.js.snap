// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdTileSelectItem renders component not selected 1`] = `
<button type="button" class="tile-select-item">
  2
</button>
`;

exports[`HdTileSelectItem renders component selected 1`] = `
<button type="button" class="tile-select-item tile-select-item--selected">
  1
</button>
`;
